hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -77,6 +79,7 @@ public String toString() {
     private volatile JobStatus jobStatus = NOT_RUNNING;
     private volatile long executionId;
     private volatile Map<MemberInfo, ExecutionPlan> executionPlanMap;
+    private volatile JobMetrics jobMetrics;
```

Comment:
I think it's possible to remove this field. The value is never reused:
- if metrics is requested for a running job, `ReportJobMetricsOp` is invoked and when all responses are received they are returned to the client
- whe `CompleteExecutionOp` returned, we can also pass the final metrics to `JobCoordinatorService.completeJob()` and not unnecessarily store it to this field.

The lifecycle of the metrics will be simpler this way.

Reply:
Let's discuss this and your comment on races when you will be available.

Reply:
This was wrong. We need the field while the job is restarting or suspended.